### PR TITLE
Add `Candidate#submission_blocked`

### PR DESCRIPTION
**Pull request by stevehook**
## Context

We want to move the `FraudMatch#blocked` attribute to `Candidate#submission_blocked` where it makes more sense (and has a clearer name). https://github.com/DFE-Digital/apply-for-teacher-training/pull/6240 added the new column to the schema. This PR completes the change.

## Changes proposed in this pull request

- [x] Data migration to copy `true` values from `FraudMatch#blocked` attribute to `Candidate#submission_blocked`.
- [x] Switch the existing support UI to use the new attribute.
- [x] Update the Candidate UI (the submission blocking part).

Not in scope:

- Dropping the old column. We'll need a follow-up migration PR to do that.

## Guidance to review

STILL DRAFT

## Link to Trello card

https://trello.com/c/1vhgJEMk/4238-add-a-submissionblocked-flag-to-candidate

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb
Comment (avinhurry) [inline spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb:46]: I like these comments!

Just wondering if this is a pattern we follow with these types of tests?

**Review by avinhurry — APPROVED** [review]
Nice work, this looks good to me. It might be worth getting a second review from @duncanjbrown (at your discretion).

File: spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb
Comment (duncanjbrown) [reply spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb:46]: Yes! I think this is great when it's not clear where we are

File: app/forms/support_interface/block_submission_form.rb
Comment (duncanjbrown) [inline app/forms/support_interface/block_submission_form.rb]: This will return false if the `Candidate` is invalid — do we want to bail under those circumstances? (Should we assume it's going to work and get an exception if something went wrong?)

File: app/forms/support_interface/block_submission_form.rb
Comment (duncanjbrown) [inline app/forms/support_interface/block_submission_form.rb:14]: ... and is it a bit weird to side-effect in an `all?` block?

File: spec/factories/course_option.rb
Comment (duncanjbrown) [inline spec/factories/course_option.rb]: Will this do the right thing if we want a course option with a different study mode?

File: spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb
Comment (duncanjbrown) [inline spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb:43]: Do we need to complete the questionnaire?

File: app/forms/support_interface/block_submission_form.rb
Comment (stevehook) [reply app/forms/support_interface/block_submission_form.rb:14]: Yes, I think the explicit `return unless` in an `each` block is easier to follow (to me) so will change...

File: spec/factories/course_option.rb
Comment (duncanjbrown) [reply spec/factories/course_option.rb]: ok great :+1:

File: spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb
Comment (stevehook) [reply spec/system/candidate_interface/references/candidate_submits_a_duplicate_application_spec.rb:43]: I just wanted to submit the application and I don't think that happens until the questionnaire is filled in.

File: app/forms/support_interface/block_submission_form.rb
Comment (stevehook) [reply app/forms/support_interface/block_submission_form.rb]: The controller handles the boolean return value although getting a `false` return value from `update` here seems unlikely.

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**
